diff --git a/ros2cli/completion/ros2-argcomplete.zsh b/ros2cli/completion/ros2-argcomplete.zsh
index b9fc309..14279de 100644
--- a/ros2cli/completion/ros2-argcomplete.zsh
+++ b/ros2cli/completion/ros2-argcomplete.zsh
@@ -20,6 +20,6 @@ autoload -U +X bashcompinit && bashcompinit
 # Get this scripts directory
 __ros2cli_completion_dir=${0:a:h}
 # Just source the bash version, it works in zsh too
-source "$__ros2cli_completion_dir/ros2-argcomplete.bash"
+source "$CONDA_PREFIX/share/bash-completion/completions/ros2-argcomplete.bash"
 # Cleanup
 unset __ros2cli_completion_dir
diff --git a/ros2cli/setup.py b/ros2cli/setup.py
index ef1ec86..9ceafb2 100644
--- a/ros2cli/setup.py
+++ b/ros2cli/setup.py
@@ -24,6 +24,17 @@ setup(
             'completion/ros2-argcomplete.fish',
             'completion/ros2-argcomplete.zsh',
         ]),
+
+        # [pixi] install autocompletion scripts
+        ('share/bash-completion/completions', [
+            'completion/ros2-argcomplete.bash',
+        ]),
+        ('share/zsh/site-functions', [
+            'completion/ros2-argcomplete.zsh',
+        ]),
+        ('share/fish/vendor_completions.d', [
+            'completion/ros2-argcomplete.fish',
+        ]),
     ],
     package_data={'': ['py.typed']},
     zip_safe=False,
